libs/platform/src/lib/components/form/time-picker/time-picker.component.ts

```diff
@@ -0,0 +1,185 @@
+     * Otherwise It can be set to a default value
+     */
+    @Input()
+    placeholder: string = this.getPlaceholder();
+
+    /** Aria label for the time picker input. */
+    @Input()
+    timePickerInputLabel = 'Time picker input';
+
+    /** Whether a null input is considered valid(success). */
+    @Input()
+    allowNull = true;
+
+    /** Defines if time component should be used with tablet mode */
+    @Input()
+    tablet = false;
+
+    /**
+     *  The placement of the popover. It can be one of: top, top-start, top-end, bottom,
+     *  bottom-start, bottom-end, right, right-start, right-end, left, left-start, left-end.
+     */
+    @Input()
+    placement: Placement = 'bottom-start';
+
+    @Input()
+    set state(state: Status) {
+        this._state = state ? state : 'default';
+    }
+
+    get state(): Status {
+        if (this.timePickerComponent && this.timePickerComponent.isInvalidTimeInput) {
+            // if any other error from core timePicker
+            return 'error';
+        }
+        if (this.status) {
+            return this.status;
+        }
+        return this._state;
+    }
+
+    /**
+     * Whether AddOn Button should be focusable, set to true by default
+     */
+    @Input()
+    buttonFocusable = true;
+
+    /**
+     * @Input when set to true, time inputs won't allow to have 1 digit
+     * for example 9 will become 09
+     * but 12 will be kept as 12.
+     */
+    @Input()
+    keepTwoDigitsTime = false;
+
+    /** Event emitted when the state of the isOpen property changes. */
+    @Output()
+    readonly isOpenChange = new EventEmitter<boolean>();
+
+    @ViewChild(TimePickerComponent)
+    timePickerComponent: TimePickerComponent;
+
+    /**
+     * @hidden
+     * The state of the form control - applies css classes.
+     * Can be `success`, `error`, `warning`, `information` or blank for default.
+     */
+    private _state: Status;
+
+    /**
+     * Method that handles changes when popover is opened or closed.
+     */
+    handleOpenChange(open: boolean): void {
+        this.isOpenChange.emit(open);
+    };
+
+    /**
+     * logic to handle validation from both platform forms and core datetiimepicker
+     * @param value inputted
+     */
+    handleTimeChange(value: TimeObject): void {
+        if (this.timePickerComponent) {
+            if (this.timePickerComponent.isInvalidTimeInput) {
+                this.state = 'error';
+            } else {
+                if (!this.timePickerComponent.time && !this.allowNull) {
+                    this.state = 'error'; // null value in not allowNull should throw error
+                } else {
+                    this.state = undefined; // resetting to default state
+                }
+            }
+            this.value = value;
+            this.stateChanges.next('time picker: handleTimeInputChange');
+        }
+        this.onTouched();
+    }
+
+    /** @hidden */
+    getPlaceholder(): string {
```

Comment:
I think it should be named `_getPlaceholder()` as it's hidden (as discussed with your team)

Reply:
I'm not a fan of this decision, but it's already discussed and approved )))))) I will change it